pkg/cli/admin/nodeimage/create_test.go

```diff
@@ -145,18 +162,26 @@ func TestRun(t *testing.T) {
 			expectedRsyncInclude: "boot-artifacts/*",
 		},
 		{
-			name:             "node-joiner tool failure",
-			nodesConfig:      defaultNodesConfigYaml,
-			objects:          defaultClusterVersionObjectFn,
-			remoteExecOutput: "1",
-			expectedError:    `image generation error: <nil> (exit code: 1)`,
+			name:        "node-joiner tool failure",
+			nodesConfig: defaultNodesConfigYaml,
+			objects:     defaultClusterVersionObjectFn,
+			remoteExecOutput: createCmdOutput(t, report{
+				stageHeader: stageHeader{
+					EndTime: time.Date(2024, 11, 14, 0, 0, 0, 0, time.UTC),
+				},
+				Result: &reportResult{
+					ExitCode:     127,
+					ErrorMessage: "Some error message",
```

Comment:
I would have expected expectedError to reflect what was reported in reportResult.ErrorMessage.

Reply:
The detailed error message is now stored in the report result section (and also logged via o.log()) (so the global outcome of the app it's just ErrExit)

Reply:
Ok thanks for the explanation. I was finding it confusing because expectedError wasn't consistent with reportResult.ErrorMessage, but I think it was just my expectation of what I should have seen from the given inputs.